lib/cAst.mli

```diff
@@ -0,0 +1,22 @@
+(************************************************************************)
+(*  v      *   The Coq Proof Assistant  /  The Coq Development Team     *)
+(* <O___,, *   INRIA - CNRS - LIX - LRI - PPS - Copyright 1999-2017     *)
+(*   \VV/  **************************************************************)
+(*    //   *      This file is distributed under the terms of the       *)
+(*         *       GNU Lesser General Public License Version 2.1        *)
+(************************************************************************)
+
+(** The ast type contains generic metadata for AST nodes *)
```

Comment:
@ejgallego Good point.

The reason why I think it is inappropriate to call `CAst.ast` as "metadata" is that it does not "full describe" AST nodes. It is "just something all AST node have". That's a difference.

If there were a corresponding (practical) mechanism in Ocaml, then `CAst.ast` would be just a supertype of all records that we attach to particular AST variants.

Supertype is not metadata.